operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -488,31 +488,38 @@ public AtomicInteger getResourceCounter() {
      * @param reconciliation reconciliation to use to update the resource state metric
      * @param ready if reconcile was successful and the resource is ready
      */
-    private void updateResourceState(Reconciliation reconciliation, boolean ready) {
-        Tags metricTags = Tags.of(
+    private void updateResourceState(Reconciliation reconciliation, boolean ready, Throwable cause) {
+        String key = reconciliation.namespace() + ":" + reconciliation.kind() + "/" + reconciliation.name();
+
+        Tags metricTagsWithReason = Tags.of(
                 Tag.of("kind", reconciliation.kind()),
                 Tag.of("name", reconciliation.name()),
-                Tag.of("resource-namespace", reconciliation.namespace()));
+                Tag.of("resource-namespace", reconciliation.namespace()),
+                Tag.of("failure-reason", cause == null ? "none" : cause.getMessage() == null ? "unknown error" : cause.getMessage()));
```

Comment:
Should the Tag be present if there is no error? What is the value of it?

Reply:
It seems that Prometheus needs the same set of tags.
```
Prometheus requires that all meters with the same name have the same set of tag keys. There is already an existing meter named 'strimzi_resource_state' containing tag keys [kind, name, reason, resource_namespace]. The meter you are attempting to register has keys [kind, name, resource_namespace]
```